nemo/collections/nlp/parts/megatron_trainer_builder.py

```diff
@@ -180,15 +182,30 @@ def _callbacks(self, callbacks: Optional[list]) -> list:
 
         return callbacks
 
+    def _profiler(self) -> Optional[Union[Profiler, str]]:
+        profiler = self.cfg.trainer.get('profiler', None)
+        del self.cfg.trainer.profiler
+        if profiler is None or isinstance(profiler, str):
+            return profiler
+        else:
+            return hydra.utils.instantiate(profiler)
+
     def create_trainer(self, callbacks=None) -> Trainer:
         # cfg.trainer.precision becomes None in Trainer if precision_plugins exist since both precision plugins and precision
         precision = self.cfg.trainer.precision
         strategy = self._training_strategy()
         plugins = self._plugins()
         callbacks = self._callbacks(callbacks)
-        trainer = Trainer(plugins=plugins, strategy=strategy, **self.cfg.trainer, callbacks=callbacks)
+        # save cfg.trainer.profiler because it will be deleted
+        original_profiler = self.cfg.trainer.get('profiler', None)
+        profiler = self._profiler()
```

Comment:
Can I burden you with adding a comment explaining what `_profiler` does?
Also re. naming, I would like to include a verb, maybe `_reinstantiate_profiler`? but up to you.

Reply:
I added a doc string for `_profiler`. I named it `_profiler` to preserve existing method style: `_training_strategy`, `_grad_scaler`, `_plugins`, `_callbacks`